tests/trainer/properties/test_estimated_num_training_steps.py

```diff
@@ -0,0 +1,167 @@
+        assert trainer.estimated_num_optimization_steps == 64
+
+    trainer = Trainer(max_epochs=1)
+    model = BoringModel()
+    trainer._data_connector.attach_data(model)
+    trainer.strategy.connect(model)
+
+    with pytest.warns(UserWarning, match="to estimate number of optimization steps"):
+        assert trainer.estimated_num_optimization_steps == 64
+
+
+def test_num_optimization_steps_iterable_dataset():
+    """Test the optimization steps with iterable dataset configured with max steps."""
+    max_steps = 1000
+    trainer = Trainer(max_steps=max_steps)
+    model = BoringModel()
+    train_dl = DataLoader(RandomIterableDataset(size=7, count=1e10))
+    trainer._data_connector.attach_data(model, train_dataloaders=train_dl)
+    trainer.strategy.connect(model)
+    assert trainer.estimated_num_optimization_steps == max_steps
+
+
+def test_num_optimization_steps_infinite_training():
+    """Test that optimization steps is "inf" when `Trainer` is configured for infinite training."""
+    trainer = Trainer(max_steps=-1, max_epochs=-1)
+    model = BoringModel()
+    trainer._data_connector.attach_data(model)
+    trainer.strategy.connect(model)
+    assert trainer.estimated_num_optimization_steps == float("inf")
+
+
+def test_num_optimization_steps_with_max_steps():
+    """Test optimization steps with `max_steps`."""
+    max_steps = 7
+    trainer = Trainer(max_steps=max_steps)
+    model = BoringModel()
+    trainer._data_connector.attach_data(model)
+    trainer.strategy.connect(model)
+    assert trainer.estimated_num_optimization_steps == max_steps
+
+
+@pytest.mark.parametrize("accumulate_grad_batches,expected_steps", [(2, 32), (3, 22)])
+def test_num_optimization_steps_accumulate_gradients(accumulate_grad_batches, expected_steps):
+    """Test the total optimization steps when accumulation grad batches is configured."""
+    trainer = Trainer(max_epochs=1, accumulate_grad_batches=accumulate_grad_batches)
+    model = BoringModel()
+    trainer._data_connector.attach_data(model)
+    trainer.strategy.connect(model)
+    assert trainer.estimated_num_optimization_steps == expected_steps
+
+
+@pytest.mark.parametrize("num_nodes,estimated_steps", [(1, 10), (2, 5), (3, 4), (4, 3)])
+def test_num_optimization_steps_ddp(num_nodes, estimated_steps, monkeypatch):
+    """Test optimization steps with DDP."""
+    monkeypatch.setattr(torch.cuda, "is_available", lambda: True)
+    monkeypatch.setattr(torch.cuda, "device_count", lambda: 7)
+    trainer = Trainer(max_epochs=1, num_nodes=num_nodes, devices=7, accelerator="gpu", strategy="ddp")
+    model = BoringModel()
+    trainer._data_connector.attach_data(model)
+    trainer.strategy.connect(model)
+    assert trainer.estimated_num_optimization_steps == estimated_steps
+
+
+@pytest.mark.parametrize("num_nodes,estimated_steps", [(1, 64), (2, 32), (3, 22)])
+def test_num_optimization_steps_ddp2(num_nodes, estimated_steps, monkeypatch):
```

Comment:
Could parameterize strategies `ddp`, `dp` and `ddp2` into one test.

Reply:
yeah.. +1